### DISCUSSION TITLE
Why?

### DISCUSSION THREAD

**Discussion by caioluis** [47ng/nuqs#495] [Q&A] [answered]
Genuine question. I'm not knowledgeable enough in state management, so although this solution seems cool, I don't know in which cases it should be used over other state management tools we have. Is it better for a particular case?

Thank you in advance!

**Comment by franky47** [comment, marked as answer]
This is a great question to ask!

Usual cases for this would be places where you'd normally use `React.useState`, but want on top of it:
- Being able to link to a particular value (sharing a URL with others)
- Bookmarking and saving in browser history
- Navigating directly from another place in your app (using `<Link>` components)

Now there are lots of types of _state_ in a web application, many of which would be unsuitable for this pattern, like server state, or local temporary UI state that does not warrant linkability.

Things that the community tends to use `nuqs` for include:
- Search (adding a `?q=my+search+input` in the URL to link back to a results page)
- Pagination (`?page=47`, using `parseAsInteger`, or time-based pagination like `?after=2024-01-01T00:00:00.000Z` using `parseAsIsoDateTime`)
- Filters for a list of items (adds linkability to a given set of filters, `parseAsStringLiteral` is particularly useful here)

**Reply by caioluis** [reply]
Alright, now I have the full picture! thanks a lot for the answer, it's clear to me when to use nuqs, i'll use it for sure.
